app/src/pages/contract/[chain]/[address].tsx

```diff
@@ -70,18 +78,32 @@ export default function Address() {
     [address]
   )
   const [constracts, setContracts] = useState<Contract[]>([])
-  const [isLoading, setIsLoading] = useState(false)
+  const [isLoadingContracts, setIsLoadingContracs] = useState(false)
+  const [addressInfos, setAddressInfos] = useState<
+    Record<string, AddressInfo[]>
+  >({})
+  const [isLoadingAddressInfos, setIsLoadingAddressInfos] = useState(false)
 
   useEffect(() => {
     if (address && chainConfig && isAddressValid) {
-      setIsLoading(true)
+      setIsLoadingContracs(true)
       fetch(`/api/address/${chain}/${address}`)
         .then((res) => res.json())
         .then((data) => {
           setContracts(data.contracts)
         })
         .finally(() => {
-          setIsLoading(false)
+          setIsLoadingContracs(false)
+        })
+
+      setIsLoadingAddressInfos(true)
+      fetch(`/api/linking/${chain}/${address}`)
+        .then((res) => res.json())
+        .then((data) => {
+          setAddressInfos(data.addresses)
+        })
+        .finally(() => {
+          setIsLoadingAddressInfos(false)
```

Comment:
I would start to fetch it as soon as the other one is _sucessfully_ finished, because:
- You might not need to fetch it (if there is no source code)
- If they're fired at the same time, they might download source code two times and that might result in duplicative records in the database (ie race condition)

Reply:
separated it :)